source/plugin/messagechannels.rst

```diff
@@ -0,0 +1,118 @@
+================
+Message Channels
+================
+
+In Sponge, every object that messages may be sent to implements the ``MessageReceiver`` interface. A ``MessageChannel``
+is a device that may send messages to an arbitrary number of ``MessageReceiver``\ s and even perform actions like
+recipient-specific formatting of the message.
+
+Selecting Message Recipients
+============================
+
+Within the ``MessageChannel`` interface there are constants and static methods that can be used to obtain or create
+commonly used channels.
+
+Broadcasting
+~~~~~~~~~~~~
+
+The most common channel will be the broadcasting channel. It can be obtained either from the ``Server`` via the
+``getBroadcastChannel()`` method or from either the ``MessageChannel.TO_PLAYERS`` or ``MessageChannel.TO_ALL`` constant.
+The only difference is that ``TO_ALL`` includes not only every player connected to the server but also the server
+``Console``.
+
+The channel returned by ``getBroadcastChannel()`` will generally be the ``MessageChannel.TO_ALL`` channel, however a
+different channel may be set using the ``setBroadcastChannel()`` method.
+
+
+Sending to a Fixed List of MessageReceivers
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+It is also possible to send a message not to all players connected, but to a number of hand-selected receivers. This
+can be done by passing the list of the intended recipients to the ``MessageChannel.fixed()`` method. Unlike most other
+channels, the list of recipients will not be generated dynamically every time something is sent through the channel but
+remains static for the rest of its existance. However, the references kept are *weak*. This means that if for example
+a player disconnects and the corresponding ``Player`` object is removed by Javas *garbage collector*, that player will
+also vanish from the channels recipient list.
+
+Permission-based Selection
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+It is also possible to create a channel sending to all recipients that hold a specific permission. The channel is
+obtained from the ``MessageChannel.permission()`` method with the permission to check for as an argument. When a
+message is then sent through this channel, it will obtain all subjects from the ``PermissionManager`` that have the
+given permission.
+
+Combining Channels
+~~~~~~~~~~~~~~~~~~
+
+It is also possible to combine multiple channels into one. This can be done by passing all channels into the
+``MessageChannel.combined()`` method. The resulting channel will relay messages to every recipient that is targeted by
+at least one of the combined channels.
+
+Sending Messages
+================
+
+Once you have obtained a ``MessageChannel`` you can send a message through it via the ``send(Text message)`` method.
+It is also possible specify a sender (which can be any ``Object``). However, that sender is only used for message
```

Comment:
The `send(Object, Text)` method should be used when possible. The `Object` is the sender, which can be used to identity and/or perform actions.